src/api/app/models/branch_package.rb

```diff
@@ -507,14 +498,8 @@ def find_packages_to_branch
       req = BsRequest.find_by_number(params[:request])
 
       req.bs_request_actions.each do |action|
-        prj=nil
-        pkg=nil
-        if action.source_project || action.source_package
-          if action.source_package
-            pkg = Package.get_by_project_and_name action.source_project, action.source_package
-          elsif action.source_project
-            prj = Project.get_by_name action.source_project
-          end
+        if action.source_package
```

Comment:
The get_by_project_and_name needs both, doesn't it? I wouldn't be to sure that both are set.